thamos/discover.py

```diff
@@ -175,15 +290,85 @@ def discover_base_image() -> typing.Optional[str]:
     return None
 
 
-def discover_all() -> typing.Dict[str, typing.Any]:
+def discover_openblas_version() -> Optional[Dict[str, str]]:
+    """Check for OpenBLAS version and return None if not found."""
+    result = run_command("rpm -qa | grep openblas", raise_on_error=False)
+
+    if result.return_code != 0:
+        _LOGGER.debug("No OpenBLAS version detected")
+        _LOGGER.debug("OpenBLAS not found in RPM packages")
+        return None
+
+    if result.stdout is not None:
+        for line in result.stdout.splitlines():
+            name, ver, rel, epoch, arch = _split_rpm_filename(line)
+            if name == "openblas":
+                _LOGGER.debug(f"OpenBLAS detected with version {ver}")
+                return {
+                    "name": name,
+                    "version": ver,
+                    "release": rel,
+                    "epoch": epoch,
+                    "architecture": arch,
+                }
+
+    return None
+
+
+def discover_openmpi_version() -> Optional[Dict[str, str]]:
+    """Check for OpenMPI version and return None if not found."""
+    result = run_command("rpm -qa | grep openmpi", raise_on_error=False)
+
+    if result.return_code != 0:
+        _LOGGER.debug("No OpenMPI version detected")
+        _LOGGER.debug("OpenMPI not found in RPM packages")
+        return None
+
+    if result.stdout is not None:
+        for line in result.stdout.splitlines():
+            name, ver, rel, epoch, arch = _split_rpm_filename(line)
+            if name == "openmpi":
+                _LOGGER.debug(f"OpenMPI detected with version {ver}")
+                return {
+                    "name": name,
+                    "version": ver,
+                    "release": rel,
+                    "epoch": epoch,
+                    "architecture": arch,
+                }
```

Comment:
Can we pull this into its own function that gets passed the library name and outputs the version?

Reply:
I think as it only concerns two test cases, we could live it as is. Otherwise we would have to adapt the log messages